### DISCUSSION TITLE
Vue Sensor values stored in ESP32 Flash?

### DISCUSSION THREAD

**Discussion by Huckcvos** [emporia-vue-local/esphome#242] [Q&A]
Hi, per Readme FAQ: "Sensor values are saved to the esp32 flash." Maybe I'm misinterpreting this statement  - are we talking about sensor configuration values or actual sensor values, which would be hundreds of values every minute? As we all know FLASH lifetime for writes to Flash are limited and if the latter would be the case then ESP32 would die a very early death, rendering the entire system non-operational.
Could someone in the know clarify please

**Comment by flaviut** [comment]
> actual sensor values, which would be hundreds of values every minute?

Correct, although they are somewhat rate limited.

> As we all know FLASH lifetime for writes to Flash are limited and if the latter would be the case then ESP32 would die a very early death, rendering the entire system non-operational.

It does seem scary, to me as well! Two things make it less scary:

1. the example config sets `restore: false` everywhere it is necessary. This means that nothing is actually written to flash
2. Flash on an embedded device like the ESP32 behaves very differently than flash in your computer. Your computer spreads out the writes evenly over the flash. The ESP32 does not do this. Writes to the saved-sensor-values partition do not impact the code partition. So even if that region of the flash was worn out, it would only break the saved sensor values feature.

**Reply by Huckcvos** [reply]
Thanks for the quick response. One more question for clarification: even with "restore:false" set, those sensor values would still be passed on to HA, or are only sensor values for which "restore:true" is set passed on the HA?

**Reply by flaviut** [reply]
`restore: false` impacts nothing except what's written to flash. [you're thinking of `internal`](https://esphome.io/components/sensor/index.html#:~:text=internal%20(Optional,%20boolean):)
